src/components/DrawerComponents.js

```diff
@@ -97,44 +97,117 @@ const useStyles = makeStyles((theme) => ({
-      to: "/user-roles",
-      icon: <FaUserCog style={{ width: "25px" }} />,
-    },
-    {
-      key: "Profile",
-      text: "My Profile",
-      to: "/profile-settings",
-      icon: <PersonIcon />,
-    },
-    {
-      key: "add-accommodation",
-      text: "Add accommodation",
-      to: "/add-accommodation",
-      icon: <AddToPhotosIcon />,
-    },
-    {
-      key: "add-rooms",
-      text: "Add rooms",
-      to: "/add-rooms",
-      icon: <PostAddIcon />,
-    },
-  ];
+  let routes;
+  const user = JSON.parse(localStorage.getItem("bn-user-data"));
+
+  if (user.role === "super_administrator") {
+    routes = [
+      {
+        key: "Dashboard",
+        text: "Dashboard",
+        to: "/dashboard",
+        icon: <HomeIcon />,
+      },
+      {
+        key: "Requests",
+        text: "My Requests",
+        to: "/requests",
+        icon: <FlightTakeoffIcon />,
+      },
+      {
+        key: "user-roles",
+        text: "Set role",
+        to: "/user-roles",
+        icon: <FaUserCog style={{ width: "25px" }} />,
+      },
+      {
+        key: "Profile",
+        text: "My Profile",
+        to: "/profile-settings",
+        icon: <PersonIcon />,
+      },
+    ];
+  } else if (
+    user.role === "travel_administrator" ||
+    user.role === "accommodation_supplier"
+  ) {
+    routes = [
+      {
+        key: "Dashboard",
+        text: "Dashboard",
+        to: "/dashboard",
+        icon: <HomeIcon />,
+      },
+      {
+        key: "Requests",
+        text: "My Requests",
+        to: "/requests",
+        icon: <FlightTakeoffIcon />,
+      },
+      {
+        key: "Profile",
+        text: "My Profile",
+        to: "/profile-settings",
+        icon: <PersonIcon />,
+      },
+      {
+        key: "add-accommodation",
+        text: "Add accommodation",
+        to: "/add-accommodation",
+        icon: <AddToPhotosIcon />,
+      },
+      {
+        key: "add-rooms",
+        text: "Add rooms",
+        to: "/add-rooms",
+        icon: <PostAddIcon />,
+      },
+    ];
+  } else if (user.role === "manager") {
+    routes = [
+      {
+        key: "Dashboard",
+        text: "Dashboard",
+        to: "/dashboard",
+        icon: <HomeIcon />,
+      },
+      {
+        key: "Requests",
+        text: "My Requests",
+        to: "/requests",
+        icon: <FlightTakeoffIcon />,
+      },
+
+      {
+        key: "Profile",
+        text: "My Profile",
+        to: "/profile-settings",
+        icon: <PersonIcon />,
+      },
+    ];
+  } else if (user.role === "requester") {
+    routes = [
+      {
+        key: "Dashboard",
+        text: "Dashboard",
+        to: "/dashboard",
+        icon: <HomeIcon />,
+      },
+      {
+        key: "Requests",
+        text: "My Requests",
+        to: "/requests",
+        icon: <FlightTakeoffIcon />,
+      },
+
+      {
+        key: "Profile",
+        text: "My Profile",
+        to: "/profile-settings",
+        icon: <PersonIcon />,
+      },
+    ];
+  }
```

Comment:
The code is redundant, make use of **destructuring assignment**

Reply:
Thanks, I have used the destructuring way.